### PR TITLE
Context Overlays

### PR DESCRIPTION
**Pull request by davidkelly**
The new behavior (when enabled) is:
- If an entity defines any context menu behavior:
  . mouse right click outlines the selected object and displays the menu item(s).
  . hovering for a short time with the half-trigger blue seek laser outlines the selected object and displays the menu item(s)
  . clicking on the menu item takes the defined action.
- The API for adding context menu behavior is TBD. This PR bakes in the "inspection" behavior: Entities with a non-empty marketplaceID have an "i" context menu item, which when activated brings up the item in the marketplace app.
- The keyboard/mouse action to bring up the tablet has been moved from right-click to the tab key.
- Apps can disable the behavior by setting `ContextOverlay.enabled=false`, and the PAL and CREATE apps do so when active.

The behavior isn't turned on yet because the outlining is currently just a yellow bounding box, which is too hard to use to be releasable.

To turn the behavior on:
- add `"inspectionMode": true` to the Interface.json in your `AppData/Roaming/High Fidelity/Interface.json`.  To turn it off, set that to false, or remove it completely.  
- Since in desktop mode, we want to inspect by using a right click of the mouse (which currently brings up the tablet), you also want to change `Program Files/High Fidelity/resources/controllers/keyboardMouse.json` so that `{ "from": "Keyboard.RightMouseClicked", "to": "Actions.ContextMenu" }` becomes `{ "from": "Keyboard.Tab", "to": "Actions.ContextMenu" }`.  Note you have to restart interface for this to take effect.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1832/HighFidelity-Beta-PR11016-5cd05da31e46ec60cfb19f1c87bff9fed460a9d3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1832/HighFidelity-Beta-PR11016-5cd05da31e46ec60cfb19f1c87bff9fed460a9d3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1839/HighFidelity-Beta-PR11016-65a0f868ce420a1d317ac8ab831db8629c7a55fb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1848/HighFidelity-Beta-PR11016-875d0eb81551182f014a3905c8821d8369750cd7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1848/HighFidelity-Beta-PR11016-875d0eb81551182f014a3905c8821d8369750cd7.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1852/HighFidelity-Beta-PR11016-9c0026bf697c95f2bb210d900b1e739a9008859c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1854/HighFidelity-Beta-PR11016-7d125f44e417d13a39e411767c2f520011820570.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1854/HighFidelity-Beta-PR11016-7d125f44e417d13a39e411767c2f520011820570.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1860/HighFidelity-Beta-PR11016-a346e5ccf6b42a6a43f45e5e29e125f91aa11b07.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1860/HighFidelity-Beta-PR11016-a346e5ccf6b42a6a43f45e5e29e125f91aa11b07.exe

**Comment by davidkelly** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1868/HighFidelity-Beta-PR11016-811be15bbcd03fe757f02a2003e9e3aeccca5b11.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1868/HighFidelity-Beta-PR11016-811be15bbcd03fe757f02a2003e9e3aeccca5b11.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1871/HighFidelity-Beta-PR11016-e7c6ba9ae2a4183d32d9542d6236adb1977cf0f5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1871/HighFidelity-Beta-PR11016-e7c6ba9ae2a4183d32d9542d6236adb1977cf0f5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1930/HighFidelity-Beta-PR11016-40038ce54b4eface3837b52fcc07aefefd9a6903.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1930/HighFidelity-Beta-PR11016-40038ce54b4eface3837b52fcc07aefefd9a6903.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1943/HighFidelity-Beta-PR11016-772372a031e090e4f49ed899a12bcf9b599882bf.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1967/HighFidelity-Beta-PR11016-7702bfceaa49cc555d39b1800ba7f9adea593a27.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1967/HighFidelity-Beta-PR11016-7702bfceaa49cc555d39b1800ba7f9adea593a27.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1992/HighFidelity-Beta-PR11016-1e80736718674aa4537a01a11fb595cbc475dc27.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1992/HighFidelity-Beta-PR11016-1e80736718674aa4537a01a11fb595cbc475dc27.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2153/HighFidelity-Beta-PR11016-bd7c4b3b261d8071a20bce6e595bffb53c92fc0f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2153/HighFidelity-Beta-PR11016-bd7c4b3b261d8071a20bce6e595bffb53c92fc0f.exe

**Comment by davidkelly** [comment]
Test Plan:

As-is (context overlays off)
- verify right click of mouse brings up the tablet (both desktop and HMD)
- rez some entities from the marketplace (flare gun, for instance)
- in HMD, use the hand controller trigger to get the 'seek' laser.  It is blue, and you get it by only partially depressing the trigger.  Blue-laser over various entities, including the stuff from the marketplace above.   Verify nothing new happens.

With context overlays on (see above)
- verify tab now brings up tablet (HMD and Desktop)
- in desktop
  - Right-click on the flare gun from above.  
  - The flare gun should be highlighted with a yellow bounding box
  - There should be an (i) icon visible
  - The (i) should pulse while the mouse cursor is not over it
  - The (i) should be solid when mouse cursor is over it
  - Click on the (i)
  - The tablet should come up, on the flare gun page of the marketplace
  - The highlighting and (i) should no longer be present (you may need to move the tablet to see this)
  - now right click on other entities you did not rez from the marketplace
  - there should be no highlighting, (i)
- in HMD
  - Blue-laser over the flare gun
  - After about 0.5 seconds hovering like that, the highlight and (i) should appear as above
  - If you keep the blue laser on, and move it off the flare gun, the (i) should pulse
  - The (i) should stay solid while your laser is 'hovering' over the flare gun
  - Click the (i) by completely depressing the trigger (you will see red laser at that point)
  - The tablet should come up as above
  - The highlighting and (i) should no longer be present
  - Now do the same over entities not rezzed from the marketplace
  - the highlight and (i) should not appear

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2163/HighFidelity-Beta-PR11016-e630e2f7e3b932b8aebdb3505e13d50cdf6aee2b.dmg

**Comment by davidkelly** [comment]
please build this

**Comment by davidkelly** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2168/HighFidelity-Beta-PR11016-e630e2f7e3b932b8aebdb3505e13d50cdf6aee2b.dmg

**Comment by davidkelly** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2177/HighFidelity-Beta-PR11016-e630e2f7e3b932b8aebdb3505e13d50cdf6aee2b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2177/HighFidelity-Beta-PR11016-e630e2f7e3b932b8aebdb3505e13d50cdf6aee2b.exe

**Comment by Nex-Pro** [comment]
Testing this.

**Comment by Nex-Pro** [comment]
### FAIL

3x fails in notes. FAIL in heavy type above failing line.

### Test Plan:

**As-is (context overlays off)**

verify right click of mouse brings up the tablet (both desktop and HMD) **SUCCESS**

rez some entities from the marketplace (several flare guns, for instance) **DONE**

in HMD, use the hand controller trigger to get the 'seek' laser. It is blue, and you get it by only partially depressing the trigger. 
Blue-laser over various entities, including the stuff from the marketplace above. 
Verify nothing new happens. **SUCCESS**

Verify using tablet (hover over buttons, press them, etc...) works as expected (desktop and HMD)**SUCCESS**


**With context overlays on (see above)**

### FAIL
Verify tab now brings up tablet (HMD and desktop) **FAIL** Tab key does not work.(Two PCs.)

Verify using tablet (hover over buttons, press them, etc...) works as expected **SUCCESS**

Go to Playa, click the Launch button, verify it launches the fireworks (HMD and desktop) **SUCCESS**


**in desktop (HMD and desktop)** ***?***

Right-click on the flare gun from above.
The flare gun should be highlighted with a yellow bounding box **SUCCESS**
There should be an (i) icon visible **SUCCESS**

The (i) should pulse while the mouse cursor is not over it **SUCCESS**
The (i) should be solid when mouse cursor is over it **SUCCESS**
Left clicking away from the entity should dismiss the highlight and (i) **SUCCESS**
Right-clicking on another marketplace entity should highlight and (i) it now  **SUCCESS**


Click on the (i)

The tablet should come up, on the flare gun page of the marketplace **SUCCESS**
The highlighting and (i) should no longer be present (you may need to move the tablet to see this) **SUCCESS**

Right click on other entities you did not rez from the marketplace
There should be no highlighting, (i) **SUCCESS**


Open the PAL (People)

Right-click on flare gun should do nothing **SUCCESS**


Close PAL (People)

Right-click should now bring up the highlight and (i) as before **SUCCESS**


Open edit.js (Create)

Right-click on flare gun should do nothing **SUCCESS**


Close edit.js (Create)
Right-click should now bring up the highlight and (i) as before **SUCCESS** also closes tablet 

### FAIL
Hit tab to bring up tablet **FAIL** Tab key does not work (Two separate PCs.)

***Invoked tablet with right click*** Which raises the question... "What happens if the whole scene, in which the avatar is located, came from the market place?" 

Verify that right-click on flare gun(s) brings up the highlight and (i) as before **SUCCESS** Easy to dismiss tablet unintentionally.
Note that clicking on tablet will dismiss the (i) and highlight **SUCCESS**

**in HMD**

Blue-laser over the flare gun
After about 0.5 seconds hovering like that, the highlight and (i) should appear as above **SUCCESS**

If you keep the blue laser on, and move it off the flare gun, the (i) should pulse **SUCCESS**

The (i) should stay solid while your laser is 'hovering' over the flare gun **SUCCESS**

The highlighting and (i) should be dismissed if you trigger over some other entity **FAIL** Can pick up another entity while (i) remains displayed on previous entity. (same hand controller) See attached image.

**Manipulating second entity while (i) remains on previous** 
hifi-snap-by--on-2017-08-02_14-20-41


The highlighting and (i) should come up again if you blue-laser over another flare gun **SUCCESS**

Click the (i) by completely depressing the trigger (you will see red laser at that point) **SUCCESS**

The tablet should come up as above **SUCCESS**

The highlighting and (i) should no longer be present **SUCCESS**


Now do the same over entities not rezzed from the marketplace

The highlight and (i) should not appear **SUCCESS**


Open the PAL (People)

Blue-lasering over flare gun should do nothing


Close PAL (People)

Blue-lasering over flare gun should bring up highlight and (i) as before **SUCCESS**


Open edit.js (Create)

### FAIL
Blue-lasering over flare gun should do nothing **FAIL** Got the (i) see attached image.



Close edit.js (Create)

Blue-lasering over flare gun should bring up highlight and (i) as before **SUCCESS**


Bring up tablet

Verify Blue-lasering over flare gun brings up highlight and (i) as before **SUCCESS**

***Note highlight and (i) remain when using tablet - bringing up menus, etc...*** 

### NB. Used two hands. Noticed that with each click on the tablet the (i) disappeared momentarily.

Verify the highlight and (i) are dismissed when bringing up PAL or edit.js **SUCCESS**

Verify you can still near/far grab marketplace items **SUCCESS**

Verify you can equip an item (like the flare gun), and blue-laser with the other hand and bring up the highlight and (i). Note the (i) will be located a bit off to the side when you are very close. **SUCCESS** Observed the (i) offset to one side and more or less within the yellow bounding box.

**Comment by Nex-Pro** [comment]
@davidkelly This is interesting. Just a couple of small issues noted.

**Comment by davidkelly** [comment]
@Nex-Pro - did you change the keyboardMouse.json as suggested above (and restart interface)?  Also I'm merging in another PR, which changes the behavior slightly, so I'll be updating the test plan...

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2202/HighFidelity-Beta-PR11016-96500d897cb3cc308ba6af73113ae824b1ce76b7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2202/HighFidelity-Beta-PR11016-96500d897cb3cc308ba6af73113ae824b1ce76b7.exe

**Comment by davidkelly** [comment]
Updated test plan to reflect the changes from @zfox23's PR.  Marked qa-ready again.

**Comment by Nex-Pro** [comment]
@davidkelly Apologies I missed that there were two jason changes... I'll do that now

**Comment by Nex-Pro** [comment]
@davidkelly Tab key is now working.  In HMD and Desktop mode

**Comment by Nex-Pro** [comment]
retesting

**Comment by Nex-Pro** [comment]
@davidkelly Did you see my question "What happens if the whole scene, in which the avatar is located, came from the market place?"

**Comment by zfox23** [comment]
@nex-pro if the camera is inside the bounding box of an entity that came from the marketplace, the overlay will show up a meter in front of the camera instead of at the edge of the bounding box.

**Comment by Nex-Pro** [comment]
@zfox23 Noted.

**Comment by Nex-Pro** [comment]
Holding awaiting @davidkelly

**Comment by davidkelly** [comment]
@Nex-Pro - Yea I'm seeing issues with this too.  Sometimes a yellow highlight appears, mostly a red one, and no (i).  I'll look into what went wrong...  Thx!

**Comment by howard-stearns** [comment]
**code review note**: An earlier version of this was a qualified pass, and it is being retested now. There are some known issues, but pending the retest, we don't think this will change.

_However_, the behavior is all turned off. Before turning it on, we will have to address some known issues (revealed in testing) that will involve changes after this merge. The issues resolve around the outline effect, which will actually be replaced by a still-to-be-developed shader.

**Comment by Nex-Pro** [comment]
Tester: Piper.Peppercorn

System information.
CPU: Intel i7 4.0 ghz.
RAM: 16gb
GPU: Nvidia GTX1070 8gb
HMD: HTC Vive.
Controllers: Vive hand controllers.
Audio: USB headset.

Testing conditions:

Fresh installation of this build.(any previous installation completely removed)


## Test Plan:

### As-is (context overlays off)

verify right click of mouse brings up the tablet (both desktop and HMD) **SUCCESS**

rez some entities from the marketplace (several flare guns, for instance) **DONE**


### in HMD:

Use the hand controller trigger to get the 'seek' laser. ***It is blue, and you get it by only partially depressing the trigger.*** Blue-laser over various entities, including the stuff from the marketplace above. Verify nothing new happens. ** Now seeing either red or yellow bounding box. Seems random **

Verify using tablet (hover over buttons, press them, etc...) works as expected (desktop and HMD) **SUCCESS**


### With context overlays on (see above)

Verify tab now brings up tablet (HMD and desktop) **SUCCESS**

Verify using tablet (hover over buttons, press them, etc...) works as expected **SUCCESS**

Go to Playa, click (in desktop) or laser click (in HMD) the Launch button, verify it launches the fireworks **SUCCESS** 


### in desktop

mouse over the flare guns, verify that they are highlighted with a yellow bounding box **Getting red bounding box**


### Right-click on one of the flare guns:

There should be an (i) icon visible **SUCCESS**

The (i) should pulse while the mouse cursor is not over it **SUCCESS**

The (i) should be solid when mouse cursor is over it **SUCCESS**

Mouse-over another flare gun should highlight it, and not change the first highlighted one with the (i) **SUCCESS**

Left clicking away from the entity should dismiss the highlight and (i) **SUCCESS**


### Mouse-over a flare gun, and right-click it

### Click on the (i)

The tablet should come up, on the flare gun page of the marketplace **SUCCESS**

All the flare guns (or other entities you rezzed from the marketplace) should be highlighted now **This is not happening for all entities from market ( three body ) ** Highlight bounding boxes are red.


### Right click on other entities you did not rez from the marketplace

 ( ammended as per David) the (i) should go away when you left click, but not when you right **SUCCESS** In addition triggering is unaffected.

Right click away from all the highlighted entities should make the (i) be dismissed, but all highlighting remains. **Highlighting remains but (i) was already dismissed by the tablet opening**

Hit x on the tablet to dismiss the marketplace, verify the highlighting goes away **SUCCESS**


### Open the PAL (People)

Mouse-over on flare gun should do nothing **FAIL** Highlights yellow.


### Close PAL (People)

Mouse-over and right-click should now bring up the highlight and (i) as before **SUCCESS**


### Open edit.js (Create)

Mouse-over on flare gun should do nothing **FAIL** Highlights yellow.


### Close edit.js (Create)

Mouse-over and right-click should now bring up the highlight and (i) as before **SUCCESS**


### Hit tab to bring up tablet

Verify that right-click on flare gun(s) brings up the highlight and (i) as before **SUCCESS**

Note clicking on various tablet buttons (like menus) dismisses highlighting and (i) **SUCCESS**


### in HMD

**Blue-laser over the flare gun**

After about 0.5 seconds hovering like that, the highlight and (i) should appear as above **SUCCESS**

If you keep the blue laser on, and move it off the flare gun, the (i) should pulse **SUCCESS**

The (i) should stay solid while your laser is 'hovering' over the flare gun **SUCCESS**

The (i) should stay on, but other flare guns should be highlighted when you laser over them **SUCCESS**

The highlighting and (i) should be dismissed if you trigger over some other entity or release the trigger completely: Trigger release **SUCCESS**  trigger over other entity **FAIL** can far-grab other entity while (i) showing on prior selection.

The highlighting and (i) should come up again if you blue-laser over another flare gun **SUCCESS**

Click the (i) by completely depressing the trigger (you will see red laser at that point)Each time 

The tablet should come up as above **Never see red laser** ( maybe it is too fast ) Tablet comes up.

The (i) is gone now, but all marketplace entities you rezzed should be highlighted **SUCCESS**

Verify that blue-lasering over one of the highlighted entities brings up the (i) (after the 0.5 sec delay) **SUCCESS**

### Dismiss the marketplace, 

verify all the highlighting goes away. **SUCCESS**


### Now blue-laser over entities not rezzed from the marketplace

The highlight and (i) should not appear **SUCCESS**


### Open the PAL (People)

Blue-lasering over flare gun should do nothing **FAIL** Highlights yellow.


### Close PAL (People)

Blue-lasering over flare gun should bring up highlight and (i) as before **SUCCESS**


### Open edit.js (Create)

Blue-lasering over flare gun should do nothing **FAIL** Highlights yellow displays (i) and triggering closes create changes tablet to vertical orientation.

hifi-snap-by--on-2017-08-02_22-59-25


### Close edit.js (Create)

Blue-lasering over flare gun should bring up highlight and (i) as before **SUCCESS**


### Bring up tablet

Verify Blue-lasering over flare gun brings up highlight and (i) as before **SUCCESS**

Note that if you use the other hand to interact with the tablet, we try (and fail somewhat) to keep a highlight and (i) on the entity your other hand is blue-lasering over. This is a dexterity test too, so we aren't concerned for now.

Verify you can still near/far grab marketplace items **SUCCESS**

Verify you can equip an item (like the flare gun), and blue-laser with the other hand and bring up the highlight and (i). Note the (i) will be located a bit off to the side when you are very close.  **SUCCESS** got the (i) in front when inspected item was equipped in either hand.


### Feedback.

This interferes with right click and drag to adjust view orientation. If my intention is to change my view I don't want to see the info icon appear on screen.

Should we be reloading the marketplace item page if it is already the right one ?

**Comment by davidkelly** [comment]
Fixed issues with pal/edit.js being up, and still seeing the highlights and so on.  Perhaps only need to retest that

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2228/HighFidelity-Beta-PR11016-4b7779f0f55381a1ea10f1b41eb6688ec7ecab69.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2228/HighFidelity-Beta-PR11016-4b7779f0f55381a1ea10f1b41eb6688ec7ecab69.exe

**Comment by Nex-Pro** [comment]
Checking the highlights.

**Comment by Nex-Pro** [comment]
@davidkelly Highlighting and(i) seem to be all good now.

**May be of interest:** 
I had a situation whereby one of the confetti guns I was using as a marketplace test object was responding to the left trigger and firing while it was not equipped. Also I was able to far grab it with right controller, manipulate the position then, while far grabbed, trigger with left controller. Pretty sure this isn't due to this PR.

PS. Remind me to paste those entries in the JSON files.

**Comment by davidkelly** [comment]
@Nex-Pro yes I've experienced that too.  I suspect it thinks it is equipped even when it is not, under some circumstances.  But I don't know how to repro it -- and seems to have been happening for awhile.

**Comment by davidkelly** [comment]
calling it qa approved per the above

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -682,7 +675,7 @@ void EntityTreeRenderer::mouseMoveEvent(QMouseEvent* event) {
 
     PickRay ray = _viewState->computePickRay(event->x(), event->y());
 
-    bool precisionPicking = false; // for mouse moves we do not do precision picking
+    bool precisionPicking = true; // for mouse moves we do precision picking
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: holy shit!! this is going to be a performance problem.

Comment (zfox23) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: how are we going to do this?

**Pull request merged**